### ISSUE TITLE
Handling of NA in DFM creation

### ISSUE THREAD

**Issue by vriezer** [quanteda/quanteda#1969]
## Describe the bug
When a vector of strings, containing one or more NAs, is converted to a DFM  using the dfm() function, the word frequencies for the missing text become 0, instead of NA. 

## Example: 
```r
library(quanteda)
text <- c("this is a very short text",NA)
dfm_example <- dfm(text)
```

## Expected behavior
Either a warning or error should be thrown, or the feature counts of the document should be NA, instead of 0

## System information

```
R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Manjaro Linux

Matrix products: default
BLAS:   /usr/lib/libblas.so.3.9.0
LAPACK: /usr/lib/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=nl_NL.UTF-8       LC_NUMERIC=C               LC_TIME=nl_NL.UTF-8        LC_COLLATE=nl_NL.UTF-8     LC_MONETARY=nl_NL.UTF-8    LC_MESSAGES=nl_NL.UTF-8    LC_PAPER=nl_NL.UTF-8      
 [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=nl_NL.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] quanteda.textmodels_0.9.1 quanteda_2.0.1           

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6       pillar_1.4.4       compiler_4.0.0     plyr_1.8.6         tools_4.0.0        stopwords_2.0      lifecycle_0.2.0    tibble_3.0.1       gtable_0.3.0       lattice_0.20-41    pkgconfig_2.0.3   
[12] rlang_0.4.6        Matrix_1.2-18      fastmatch_1.1-0    rstudioapi_0.11    RSSL_0.9.1         SparseM_1.78       cluster_2.1.0      stringr_1.4.0      LiblineaR_2.10-8   dplyr_1.0.0        generics_0.0.2    
[23] vctrs_0.3.0        fs_1.4.1           grid_4.0.0         tidyselect_1.1.0   glue_1.4.1         data.table_1.12.8  R6_2.4.1           kernlab_0.9-29     ggplot2_3.3.0      purrr_0.3.4        reshape2_1.4.4    
[34] magrittr_1.5       MASS_7.3-51.6      scales_1.1.0       ellipsis_0.3.0     usethis_1.6.0      colorspace_1.4-1   quadprog_1.5-8     stringi_1.4.6      RcppParallel_5.0.0 munsell_0.5.0      crayon_1.3.4   
```

**Comment by kbenoit** [comment]
Not a bug, but a "feature"!

An empty document has zero counts for the features contained in the corpus, which in your example, come entirely from those found in document 1. Document 2 has 0 counts because it contains zero occurrences of those features.

```r
> text <- c("this is a very short text",NA)
> dfm_example <- dfm(text)
> dfm_example
Document-feature matrix of: 2 documents, 6 features (50.0% sparse).
       features
docs    this is a very short text
  text1    1  1 1    1     1    1
  text2    0  0 0    0     0    0
```
is intended, in other words.

**Comment by vriezer** [comment]
Semantically this is not true, NA denotes something different than a 0 count. While an empty string indeed contains 0 features, an NA is a missing value, and thus it is not known whether the feature count is 0. The reason I care about this difference is that predict.textmodel_nb(), as a result, predicts classes for NA documents, which it should not, because in truth there is no information to base the prediction on.

**Comment by kbenoit** [comment]
Probably best then to filter out the documents with zero features. Or to remove documents that are NA.

@koheiw might have some thoughts on this, but if we did handle this specially, it would need to be at the `tokens()` construction stage. A `tokens` object needs to be able to contain zero tokens since `tokens_select()` can easily result in zero-length tokens in a document. We don’t currently allow any document to have `NA` in either a tokens or a corpus object. 

So @koheiw if we did handle this specially, it would be that `corpus(NA)` removes the NA document or produces empty tokens, perhaps according to the value of a new argument. For a zero-length tokens to result, `corpus(“”)` will continue to do that.
